dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceOutputStream.java

```diff
@@ -182,26 +203,64 @@ public class RowServiceOutputStream extends OutputStream
             throw new Exception(errorMessage);
         }
 
+        //------------------------------------------------------------------------------
+        // Check protocol version
+        //------------------------------------------------------------------------------
+
+        try
+        {
+            String msg = makeGetVersionRequest();
+            int msgLen = msg.length();
+
+            this.dos.writeInt(msgLen);
```

Comment:
There would be a slight increase in connect time, but it should be pretty close to the ping time. So on internal networks it would only add < 10ms? Which I think it negligible

Reply:
No, it is the same end-point. Yes, the legacy server does respond to that version request, but essentially with an "I don't understand this request response." Which is what I am checking for to determine which protocol to use. Jake explicitly designed that version request so that the legacy server would be able to respond with the I don't understand response.